CMakeLists.txt

```
@@ -1,4 +1,10 @@
-cmake_minimum_required(VERSION 3.12)
+option(TMP_MODULES "Build C++20 modules" OFF)
+if(TMP_MODULES)
+  cmake_minimum_required(VERSION 3.28)
+else()
+  cmake_minimum_required(VERSION 3.12)
+endif()
```

Per [CMake documentation](https://cmake.org/cmake/help/latest/command/cmake_minimum_required.html):
```
Call the cmake_minimum_required() command at the beginning of the top-level CMakeLists.txt file even before calling the [project()](https://cmake.org/cmake/help/latest/command/project.html#command:project) command. It is important to establish version and policy settings before invoking other commands whose behavior they may affect.
```

```suggestion
cmake_minimum_required(VERSION 3.12)

option(TMP_MODULES "Build C++20 modules" OFF)
if(TMP_MODULES)
  cmake_minimum_required(VERSION 3.28)
endif()
```